R-package/R/lgb.Booster.R

```diff
@@ -828,23 +815,12 @@ predict.lgb.Booster <- function(object,
                                 predcontrib = FALSE,
                                 header = FALSE,
                                 reshape = FALSE,
-                                params = list(),
-                                ...) {
+                                params = list()) {
```

Comment:
It looks like `...` cannot be removed from `predict.lgb.Booster()`, since the `predict()` generic in base R has a `...`.

From `R CMD check` ([build link](build link: https://github.com/microsoft/LightGBM/runs/4420930711?check_suite_focus=true))

```text
* checking S3 generic/method consistency ... WARNING
predict:
  function(object, ...)
predict.lgb.Booster:
  function(object, data, start_iteration, num_iteration, rawscore,
           predleaf, predcontrib, header, reshape, params)

See section ‘Generic functions and methods’ in the ‘Writing R
Extensions’ manual.
```

From https://cran.r-project.org/doc/manuals/R-exts.html#Generic-functions-and-methods

> A method must have all the arguments of the generic, including … if the generic does.

Reply:
Add a warning and updated docs in https://github.com/microsoft/LightGBM/pull/4857/commits/49c8368124f57a5946b226043052efc10faa316e, so that at least users are notified that `...` is not intended for use with any `{lightgbm}` parameters.